packages/tupaia-web/src/components/Modal.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React, { ReactNode } from 'react';
+import { Dialog, Paper as MuiPaper, useTheme, useMediaQuery } from '@material-ui/core';
+import MuiCloseIcon from '@material-ui/icons/Close';
+import styled from 'styled-components';
+import { Button } from '@tupaia/ui-components';
+
+interface ModalProps {
+  children?: ReactNode;
+  onClose: () => void;
+  isOpen: boolean;
+}
+
+const Wrapper = styled.div`
+  text-align: center;
+  overflow-x: hidden;
+  padding: 2em;
+`;
+
+const CloseIcon = styled(MuiCloseIcon)`
+  width: 1.2em;
+  height: 1.2em;
+`;
+
+const CloseButton = styled(Button)`
```

Comment:
```suggestion
const CloseButton = styled(IconButton)`
```
Can we use the IconButton as the base component?